worlds/jakanddaxter/Client.py

```diff
@@ -0,0 +1,601 @@
+                        f"   Close all launchers, games, clients, and console windows, then restart Archipelago.")
+
+    if not os.path.exists(settings_path):
+        msg = (f"{err_title}: the OpenGOAL settings file does not exist.\n"
+               f"{alt_instructions}")
+        ctx.on_log_error(logger, msg)
+        return
+
+    with open(settings_path, "r") as f:
+        load = json.load(f)
+
+        jak1_installed = load["games"]["Jak 1"]["isInstalled"]
+        if not jak1_installed:
+            msg = (f"{err_title}: The OpenGOAL Launcher is missing a normal install of Jak 1!\n"
+                   f"{alt_instructions}")
+            ctx.on_log_error(logger, msg)
+            return
+
+        mod_sources = load["games"]["Jak 1"]["modsInstalledVersion"]
+        if mod_sources is None:
+            msg = (f"{err_title}: No mod sources have been configured in the OpenGOAL Launcher!\n"
+                   f"{alt_instructions}")
+            ctx.on_log_error(logger, msg)
+            return
+
+        # Mods can come from multiple user-defined sources.
+        # Make no assumptions about where ArchipelaGOAL comes from, we should find it ourselves.
+        archipelagoal_source = None
+        for src in mod_sources:
+            for mod in mod_sources[src].keys():
+                if mod == "archipelagoal":
+                    archipelagoal_source = src
+                    # Using this file, we could verify the right version is installed, but we don't need to.
+        if archipelagoal_source is None:
+            msg = (f"{err_title}: The ArchipelaGOAL mod is not installed in the OpenGOAL Launcher!\n"
+                   f"{alt_instructions}")
+            ctx.on_log_error(logger, msg)
+            return
+
+        # This is just the base OpenGOAL directory, we need to go deeper.
+        base_path = load["installationDir"]
+        mod_relative_path = f"features/jak1/mods/{archipelagoal_source}/archipelagoal"
+        mod_path = os.path.normpath(
+            os.path.join(
+                os.path.normpath(base_path),
+                os.path.normpath(mod_relative_path)))
+
+    return mod_path
+
+
+async def run_game(ctx: JakAndDaxterContext):
+
+    # These may already be running. If they are not running, try to start them.
+    # TODO - Support other OS's. 1: Pymem is Windows-only. 2: on Linux, there's no ".exe."
+    gk_running = False
+    try:
+        pymem.Pymem("gk.exe")  # The GOAL Kernel
+        gk_running = True
+    except ProcessNotFound:
+        ctx.on_log_warn(logger, "Game not running, attempting to start.")
+
+    goalc_running = False
+    try:
+        pymem.Pymem("goalc.exe")  # The GOAL Compiler and REPL
+        goalc_running = True
+    except ProcessNotFound:
+        ctx.on_log_warn(logger, "Compiler not running, attempting to start.")
+
+    try:
+        settings = Utils.get_settings()
+        auto_detect_root_directory = settings["jakanddaxter_options"]["auto_detect_root_directory"]
```

Comment:
please use the settings api properly,
```suggestion
        settings = settings.get_settings()
        auto_detect_root_directory = settings.jakanddaxter_options.auto_detect_root_directory
```

Reply:
Fixed both this and one instance of "Utils.get_settings" in \_\_init\_\_.

Reply:
follow up in case you didn't see the convo in discord, you can also (and probably should) use `JakAndDaxterWorld.settings.auto_detect_root_directory` instead